### PR TITLE
dummy video source support in e2e test

### PR DESCRIPTION
**Pull request by edmond-zhu**
1. Add dummy video source (color bar) support in e2e test.

### CODE REVIEW

**Comment by edmond-zhu** [comment]
# For dummy video source support of e2e test:

# Pr for add dummy video source support in e2e test
## license: check if license is ok
## Comments
## warnings in Local build
1. mac: no new warning, only warning: -pie being ignored. It is only used when linking a main executable
2. linux: GNU warning in SpatialFullDistortionTest.cc
3. windows: none

## warnings on remote ci servers
1. jenkins:
a. clang-format
b. 
2. travis
  a. mac: `-pie being ignored. It is only used when linking a main executable`, can be ignored.
  b. linux: SpaticalFullDistoration wanring, could be fixed by merge the latest code;
3. appvedor
  a. no warnings

## dry run output
~~~
Running main() from C:\Works\AV1\SVT-AV1\third_party\googletest\src\gtest_main.cc
Note: Google Test filter = */ConformanceDeathTest.*/DummySrcTest1
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SvtAv1/ConformanceDeathTest
[ RUN      ] SvtAv1/ConformanceDeathTest.DefaultSettingTest/DummySrcTest1
Start test case DummySrcTest1: EncoderMode=8, test vector: colorbar_480p_8_420
Running main() from C:\Works\AV1\SVT-AV1\third_party\googletest\src\gtest_main.cc
Start test case DummySrcTest1: EncoderMode=8, test vector: colorbar_480p_8_420
SVT [version]:  SVT-AV1 Encoder Lib v0.6.0
SVT [build]  :  Visual Studio 2017 64 bit
LIB Build date: Jul 18 2019 11:41:30
-------------------------------------------
Number of logical cores available: 12
Number of PPCS 72
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                       : 8
SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat     : 8 / 1 / 0
SVT [config]: SourceWidth / SourceHeight                                        : 640 / 480
SVT [config]: FrameRate / Gop Size                                              : 30 / 32
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure          : 4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                  : CQP / 50 / 33 / 0
-------------------------------------------
SVT Memory Usage:
    total allocated memory:       1.61 GB
        malloced memory:          1.19 GB
        callocated memory:        36.60 MB
        allocated aligned memory: 396.30 MB
    mutex count: 573
    semaphore count: 345
    thread count: 75
    hash table fulless: 0.032791, hash bucket is healthy
top 10 malloced memory locations:
(227.86 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbPictureControlSet.c:1201
(144.96 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Encoder\Codec\EbEncHandle.c:3430
(143.28 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbModeDecisionProcess.c:91
(143.05 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbBitstreamUnit.c:40
(88.76 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbRestoration.c:1737
(43.59 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbCodingUnit.c:60
(37.64 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbModeDecisionProcess.c:92
(37.50 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbCodingUnit.c:61
(35.16 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbMotionEstimationContext.c:117
(35.16 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbMotionEstimationContext.c:118
Encoder EOS
PSNR: 100 frames, total: 51.6080, luma: 49.3074, cb: 44.7943, cr: 60.7224
Enc Performance: 0.02sec/frame (50.9944FPS)
SVT: you have no memory leak
[       OK ] SvtAv1/ConformanceDeathTest.DefaultSettingTest/DummySrcTest1 (9307 ms)
[----------] 1 test from SvtAv1/ConformanceDeathTest (9308 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (9310 ms total)
[  PASSED  ] 1 test.

~~~

File: test/e2e_test/VideoSource.cc [hunk 1]
```diff
@@ -30,7 +30,10 @@ VideoSource::VideoSource(const VideoColorFormat format, const uint32_t width,
       current_frame_index_(-1),
```
Comment (Cidana-Developers) [inline test/e2e_test/VideoSource.cc:30]: The CL in VideoSource does not related with dummy video source support, maybe put it in another pr.

File: test/e2e_test/DummyVideoSource.h [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                     const bool use_compressed_2bit_plan_output)
+        : VideoSource(format, width, height, bit_depth,
+                      use_compressed_2bit_plan_output) {
+        if (width_ % 16 != 0) {
+            width_with_padding_ = ((width_ >> 4) + 1) << 4;
+        }
+        if (height_ % 16 != 0) {
+            height_with_padding_ = ((height_ >> 4) + 1) << 4;
+        }
+#ifdef ENABLE_DEBUG_MONITOR
```
Comment (Cidana-Developers) [inline test/e2e_test/DummyVideoSource.h]: Should we remove ENABLE_DEBUG_MONITOR? The building is broken now.

**Comment by FochQu** [comment]
test stream vector

**Review by Cidana-Developers — APPROVED** [review]

**Comment by Cidana-Developers** [comment]
@edmond-zhu please add Test Plan items to show how to run it so Foch can verify it.

**Comment by edmond-zhu** [comment]
For the test plan, please refer to the dry run log:
SvtAv1UnitTests --gtest_filter=*/ConformanceDeathTest.*/DummySrcTest1

**Review by Cidana-Developers — APPROVED** [review]

**Comment by Cidana-Developers** [comment]
Merged.
